src/lib.rs

```diff
@@ -49,6 +49,8 @@ mod waker;
 pub mod event;
 
 cfg_net! {
```

Comment:
`io_souce` is only used with the UDP feature for this change. Therefore, to prevent warnings on builds such as `cargo check --features tcp`, this should be changed to:

```
cfg_net! {
    pub mod net;
}

// `io_source` is only used by the UDP feature. This will be changed to
// `cfg_net!` once the other net resources use `io_source`.
cfg_udp! {
    mod io_source;
}
```

Reply:
I know but I intend to switch all net types to use `IoSource`, e.g. see #1196, so I took a bit of a shortcut to prefer future merge conflicts.